multi-agent/lib/mas/core/channels/protocols.py

```
@@ -107,6 +139,26 @@ def create(self, session_id: str) -> SessionChannel:
         """Create a write channel for the given session."""
         ...
 
+    def create_input_capable(self, session_id: str) -> Optional[InputCapableChannel]:
+        """Create a bidirectional channel for HITL.
+
+        Returns ``None`` when the backend does not support inbound
+        input.  Override in subclasses that provide HITL support.
+        """
+        return None
+
+    def get_input_channel(self, session_id: str) -> Optional[Any]:
```

Ruff (0.15.20)

[warning] Unused method argument: `session_id`

(ARG002)